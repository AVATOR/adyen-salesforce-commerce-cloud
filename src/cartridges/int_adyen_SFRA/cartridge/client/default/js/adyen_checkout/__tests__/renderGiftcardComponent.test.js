/**
 * @jest-environment jsdom
 */
const {
          removeGiftCard,
          renderAddedGiftCard,
          renderGiftCardSelectForm,
          getGiftCardElements,
          showGiftCardWarningMessage,
          createElementsToShowRemainingGiftCardAmount } = require('../renderGiftcardComponent');
const { createSession } = require('../../commons');
const store = require('../../../../../store');

jest.mock('../../commons');
jest.mock('../../../../../store');

beforeEach(() => {
  window.AdyenCheckout = jest.fn(async () => ({
    create: jest.fn(),
    paymentMethodsResponse: {
      storedPaymentMethods: [{ supportedShopperInteractions: ['Ecommerce'] }],
      paymentMethods: [{ type: 'amazonpay' }],
    },
    options: {
      amount: 'mocked_amount',
      countryCode: 'mocked_countrycode',
    }
  }));
  window.Configuration = { amount: 0 };
  store.checkoutConfiguration = {
    session: {
        imagePath: "test_image_path"
    }
  };
  store.checkout = {
   options: {}
  };
  createSession.mockReturnValue({
    adyenConnectedTerminals: { uniqueTerminalIds: ['mocked_id'] },
    id: 'mock_id',
    sessionData: 'mock_session_data',
    imagePath: 'example.com',
    adyenDescriptions: {},
  });
});
describe.only('Render gift card', () => {
  it('should showGiftCardWarningMessage', async () => {
    document.body.innerHTML = `
      <div class="card-body order-total-summary"></div>
    `;

    showGiftCardWarningMessage();
      const orderTotalSummaryEl = document.querySelector(
        '.card-body.order-total-summary',
      );
    expect(orderTotalSummaryEl).toMatchSnapshot();
    expect(document.querySelector('#giftCardWarningMessage')).toBeTruthy();
  });

    it('should renderAddedGiftCard', async () => {
      store.partialPaymentsOrderObj = {giftcard: {brand: 'brand'}};
      document.body.innerHTML = `
        <select id="giftCardSelect"></select>
        <ul id="giftCardUl"></ul>
        <ul id="giftCardsList"></ul>
        <div id="giftCardContainer"></div>
        <div id="giftCardSelectContainer"></div>
        <button id="giftCardAddButton"></button>
      `;

      renderAddedGiftCard();

        const giftCardAddButton = document.querySelector('#giftCardAddButton');
        const giftCardsList = document.querySelector('#giftCardsList');
        expect(giftCardsList).toMatchSnapshot();
//      brandAndRemoveActionWrapper.classList.add('wrapper');
      expect(document.querySelector('#giftCardAddButton').style.display).toBe('none');
    });
});
