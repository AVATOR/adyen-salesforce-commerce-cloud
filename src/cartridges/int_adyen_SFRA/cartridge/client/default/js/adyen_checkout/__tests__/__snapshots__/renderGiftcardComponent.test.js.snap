// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render gift card should renderAddedGiftCard 1`] = `
<ul
  id="giftCardsList"
>
  <div
    class="gift-card"
  >
    <div
      class="wrapper"
    >
      <div
        class="brand-container"
      >
        <img
          class="gift-card-logo"
          src="test_image_pathbrand.png"
        />
        <p />
      </div>
      <div
        class="gift-card-action"
      >
        <a />
      </div>
    </div>
    <div
      class="wrapper"
    >
      <p />
      <strong />
    </div>
  </div>
</ul>
`;

exports[`Render gift card should showGiftCardWarningMessage 1`] = `
<div
  class="card-body order-total-summary"
>
  <div
    class="alert alert-warning error-message gift-card-warning-msg"
    id="giftCardWarningMessage"
    role="alert"
  >
    <p
      class="error-message-text"
    />
  </div>
</div>
`;
